h2/stream.py

```diff
@@ -1020,17 +1033,36 @@ def stream_reset(self, frame):
 
         return [], events
 
+    def _build_hdr_validation_flags(self, events):
+        """
+        Constructs a set of header validation flags for use when normalizing
+        and validating header blocks.
+        """
+        try:
+            is_trailer = isinstance(events[0], _TrailersSent)
```

Comment:
Let's add `TrailersReceived` to the list of trailery events, because we can then use this method in `receive_headers` too.